.pre-commit-config.yaml

```
@@ -51,6 +51,8 @@ repos:
     - commit
     - manual
   - id: no-commit-to-branch
+    # protect main and any branch that has a semver-like name
+    args: [-b, main, -p, '^\d+.\d+(?:.\d)*$']
```

The dot needs to be escaped to not match any character. 
We also don't have more than three versions
Can we just: 
```suggestion
    args: [-b, main, -p, '^\d+\.\d+(?:\.\d+)?$']
```
for 2.4 / 244.35235 / 2.4.1?

---

Thanks, I forgot the dot was a special character too. 
The regex you proposed was exactly what I intended, I just wrote it down incorrectly.

---

amended the commit